clients/spark/core/src/main/scala/io/treeverse/gc/CommittedAddressLister.scala

```diff
@@ -26,7 +26,10 @@ class NaiveCommittedAddressLister extends CommittedAddressLister {
       .withColumn("absolute_address", concat(lit(normalizedStorageNamespace), df("address")))
     // TODO push down a filter to the input format, to filter out absolute addresses!
     df = df
-      .select(df("absolute_address").as("address"))
+      // TODO (niro): Revert substring after https://github.com/treeverse/lakeFS/issues/4699
+      .select(df("absolute_address"),
+              substring_index(col("absolute_address"), "/", -3).as("address")
```

Comment:
Why hardcoding `-3`? Storage namespace can be 
1. `s3://bucket/`
2. `s3://bucket/path/`
3. `s3://bucket/some/very/long/path/`
 
And I think we handle it, but we need to take into account storage namespace that does or doesn't end with a slash.

Reply:
We have the storage namespace, let's use it :)